esp-hal-common/src/uart.rs

```diff
@@ -189,6 +189,18 @@ pub mod config {
             }
         }
     }
+
+    /// Configuration of the receive timeout functionality
+    pub enum RxTimeoutConfig {
+        /// Disable receive timeout
+        Disabled,
+        /// Timeout after that many bits. Maximum value: 0x7F (7 bits)
+        #[cfg(not(esp32))]
+        TimeoutBits(u16),
```

Comment:
We should abstract over this so its the same API per chip. Imo we should take the argument as bits, and on the esp32 divide by 8 to get the value as bytes.